sale_purchase_history/models/product_sale_history.py

```diff
@@ -0,0 +1,19 @@
+from odoo import api, fields, models
+
+class ProductSaleHistory(models.TransientModel):
+    _name = 'product.sale.history'
+
+    product_id = fields.Many2one('product.product')
+    is_sale = fields.Boolean(default=lambda self: self.env.context.get('is_sale', False))
+    is_purchase = fields.Boolean(default=lambda self: self.env.context.get('is_purchase', False))
```

Comment:
this should be done in default_get

or even better, when calling the wizard, add context "default_is_sale": true or "default_is_true": true, this is a standard way of setting default values in newly created records

 but you don't need the fields anyway because you can check the context in the compute function directly